engines/agi/saveload.cpp

```diff
@@ -85,11 +85,19 @@ int AgiEngine::saveGame(const Common::String &fileName, const Common::String &de
 
 	out->writeUint32BE(AGIflag);
 
+	const char *descriptionStringC;
+	if (_game._vm->getLanguage() != Common::HE_ISR) {
+		descriptionStringC = descriptionString.c_str();
+	} else {
+		Common::U32String u32string = descriptionString.decode(Common::kWindows1255);
+		descriptionStringC = u32string.encode(Common::kUtf8).c_str();
```

Comment:
You keep a pointer to a variable that gets out of scope. Bad practice, especially if the compiler reuses the stack for other scopes (default with gcc).

You can do something like this:
```
const Common::String *description = &descriptionString;
Common::String hebDesc;
if (_game._vm->getLanguage() == Common::HE_ISR) {
    hebDesc = descriptionString.decode(Common::kWindows1255).encode();
    description = &hebDesc;
}
```

Reply:
Nice catch!
Thanks

Reply:
I've fixed the scope problem, in another way.